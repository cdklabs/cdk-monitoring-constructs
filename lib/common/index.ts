export * from "./alarm";
export * from "./metric";
export * from "./monitoring";
export * from "./url";
export * from "./widget";
export * from "./strings";
