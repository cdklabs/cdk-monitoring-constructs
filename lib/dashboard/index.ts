export * from "./BitmapDashboard";
export * from "./DashboardRenderingPreference";
export * from "./DynamicDashboardSegment";
export * from "./DashboardSegment";
export * from "./DashboardWithBitmapCopy";
export * from "./DefaultDashboardFactory";
export * from "./DynamicDashboardFactory";
export * from "./DefaultWidgetFactory";
export * from "./IDashboardFactory";
export * from "./IDynamicDashboardFactory";
export * from "./IWidgetFactory";
export * from "./MonitoringNamingStrategy";
export * from "./SingleWidgetDashboardSegment";
export * from "./widget";
