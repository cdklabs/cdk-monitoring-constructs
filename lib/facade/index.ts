export * from "./IMonitoringAspect";
export * from "./MonitoringFacade";
export * from "./MonitoringAspect";
