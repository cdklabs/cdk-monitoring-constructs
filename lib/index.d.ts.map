{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["index.ts"],"names":[],"mappings":"AAAA,cAAc,UAAU,CAAC;AACzB,cAAc,aAAa,CAAC;AAC5B,cAAc,UAAU,CAAC;AACzB,cAAc,cAAc,CAAC"}