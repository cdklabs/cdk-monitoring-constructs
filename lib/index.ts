export * from "./common";
export * from "./dashboard";
export * from "./facade";
export * from "./monitoring";
