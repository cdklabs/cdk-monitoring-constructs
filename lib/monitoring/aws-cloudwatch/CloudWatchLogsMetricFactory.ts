import { DimensionsMap } from "aws-cdk-lib/aws-cloudwatch";

import {
  BaseMetricFactory,
  BaseMetricFactoryProps,
  MetricFactory,
  MetricStatistic,
} from "../../common";

const CloudWatchLogsNamespace = "AWS/Logs";

export interface CloudWatchLogsMetricFactoryProps
  extends BaseMetricFactoryProps {
  /**
   * Name of the log group to monitor.
   */
  readonly logGroupName: string;
}

export class CloudWatchLogsMetricFactory extends BaseMetricFactory<CloudWatchLogsMetricFactoryProps> {
  private readonly dimensionsMap: DimensionsMap;

  constructor(
    metricFactory: MetricFactory,
    props: CloudWatchLogsMetricFactoryProps,
  ) {
    super(metricFactory, props);

    this.dimensionsMap = {
      LogGroupName: props.logGroupName,
    };
  }

  metricIncomingLogEvents() {
    return this.metricFactory.createMetric(
      "IncomingLogEvents",
      MetricStatistic.N,
      "Logs",
      this.dimensionsMap,
      undefined,
      CloudWatchLogsNamespace,
      undefined,
      this.region,
      this.account,
    );
  }
}
