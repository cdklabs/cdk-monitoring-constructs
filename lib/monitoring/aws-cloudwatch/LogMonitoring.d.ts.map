{"version":3,"file":"LogMonitoring.d.ts","sourceRoot":"","sources":["LogMonitoring.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,EAAkB,MAAM,4BAA4B,CAAC;AAErE,OAAO,EACL,mBAAmB,EAGnB,UAAU,EACV,eAAe,EAChB,MAAM,cAAc,CAAC;AAKtB,MAAM,WAAW,kBAAmB,SAAQ,mBAAmB;IAC7D;;OAEG;IACH,QAAQ,CAAC,YAAY,EAAE,MAAM,CAAC;IAC9B;;OAEG;IACH,QAAQ,CAAC,OAAO,EAAE,MAAM,CAAC;IACzB;;;OAGG;IACH,QAAQ,CAAC,KAAK,CAAC,EAAE,MAAM,CAAC;IACxB;;;OAGG;IACH,QAAQ,CAAC,KAAK,CAAC,EAAE,MAAM,CAAC;CACzB;AAED;;GAEG;AACH,qBAAa,aAAc,SAAQ,UAAU;IAC3C,SAAS,CAAC,QAAQ,CAAC,OAAO,EAAE,MAAM,CAAC;IACnC,SAAS,CAAC,QAAQ,CAAC,YAAY,EAAE,MAAM,CAAC;IACxC,SAAS,CAAC,QAAQ,CAAC,WAAW,CAAC,EAAE,MAAM,CAAC;IACxC,SAAS,CAAC,QAAQ,CAAC,KAAK,CAAC,EAAE,MAAM,CAAC;IAClC,SAAS,CAAC,QAAQ,CAAC,KAAK,EAAE,MAAM,CAAC;gBAErB,KAAK,EAAE,eAAe,EAAE,KAAK,EAAE,kBAAkB;IAY7D,OAAO,IAAI,OAAO,EAAE;IA0BpB,SAAS,CAAC,wBAAwB,CAAC,OAAO,EAAE,MAAM;CAKnD"}