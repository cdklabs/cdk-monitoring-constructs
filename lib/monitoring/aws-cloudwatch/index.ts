export * from "./CloudWatchLogsMetricFactory";
export * from "./LogMonitoring";
