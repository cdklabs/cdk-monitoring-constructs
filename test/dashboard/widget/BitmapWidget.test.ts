import { Stack } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { GraphWidget } from "aws-cdk-lib/aws-cloudwatch";

import { BitmapWidgetRenderingSupport } from "../../../lib";
import { forceStableAssetKeys } from "../../utils/StableTestKeys";

test("support", () => {
  const stack = new Stack();

  new BitmapWidgetRenderingSupport(stack, "Support");

  forceStableAssetKeys(stack);

  expect(Template.fromStack(stack)).toMatchSnapshot();
});

test("widget", () => {
  const stack = new Stack();

  const support = new BitmapWidgetRenderingSupport(stack, "Support");

  const widget = support.asBitmap(
    new GraphWidget({
      width: 12,
      height: 6,
      title: "Dummy Title",
      left: [],
      leftAnnotations: [{ label: "Dummy Annotation", value: 10 }],
    }),
  );

  expect(widget.toJson()).toEqual([
    {
      height: 6,
      properties: {
        endpoint: support.handler.functionArn,
        params: {
          graph: {
            annotations: {
              horizontal: [
                {
                  label: "Dummy Annotation",
                  value: 10,
                  yAxis: "left",
                },
              ],
            },
            legend: undefined,
            liveData: undefined,
            metrics: undefined,
            period: undefined,
            region: stack.region,
            setPeriodToTimeRange: undefined,
            stacked: undefined,
            stat: undefined,
            title: " ",
            view: "timeSeries",
            yAxis: {
              left: undefined,
              right: undefined,
            },
          },
        },
        title: "Dummy Title",
        updateOn: {
          refresh: true,
          resize: true,
          timeRange: true,
        },
      },
      type: "custom",
      width: 12,
      x: undefined,
      y: undefined,
    },
  ]);
});
