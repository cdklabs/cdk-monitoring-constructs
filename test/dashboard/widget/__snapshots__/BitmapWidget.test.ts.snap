// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`support 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A": Object {
      "DependsOn": Array [
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "Fixed key that does not depend on asset changes",
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:PutRetentionPolicy",
                "logs:DeleteRetentionPolicy",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "Roles": Array [
          Object {
            "Ref": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "SupportLambdaE954141B": Object {
      "DependsOn": Array [
        "SupportLambdaServiceRoleDefaultPolicyF2B2E1DB",
        "SupportLambdaServiceRole250832F3",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "Fixed key that does not depend on asset changes",
        },
        "Description": "Custom Widget Render for Bitmap Widgets (MonitoringCDKConstructs)",
        "Handler": "index.handler",
        "MemorySize": 128,
        "Role": Object {
          "Fn::GetAtt": Array [
            "SupportLambdaServiceRole250832F3",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
        "Tags": Array [
          Object {
            "Key": "cw-custom-widget",
            "Value": "describe:readOnly",
          },
        ],
        "Timeout": 60,
      },
      "Type": "AWS::Lambda::Function",
    },
    "SupportLambdaLogRetention522A4BC8": Object {
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/lambda/",
              Object {
                "Ref": "SupportLambdaE954141B",
              },
            ],
          ],
        },
        "RetentionInDays": 1,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
            "Arn",
          ],
        },
      },
      "Type": "Custom::LogRetention",
    },
    "SupportLambdaServiceRole250832F3": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": Array [
          Object {
            "Key": "cw-custom-widget",
            "Value": "describe:readOnly",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "SupportLambdaServiceRoleDefaultPolicyF2B2E1DB": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "cloudwatch:GetMetricWidgetImage",
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "SupportLambdaServiceRoleDefaultPolicyF2B2E1DB",
        "Roles": Array [
          Object {
            "Ref": "SupportLambdaServiceRole250832F3",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
